Assets/_Scripts/RewardScene/RewardSceneController.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class RewardSceneController : MonoBehaviour
+{
+
+    public AffinityController affinityController;
+    public RewardController rewardController;
+
+    public Transform affinityParent;
+    public Transform rewardParent;
+
+
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        this.affinityParent.gameObject.SetActive(false);
+        this.rewardParent.gameObject.SetActive(false);
+
+        this.InitializeAffinityController();
+    }
+
+    public void InitializeAffinityController() {
+        this.affinityParent.gameObject.SetActive(true);
+        affinityController.Initialize(this.InitializeRewardController);
+    }
+
+    public void InitializeRewardController() {
+        this.affinityParent.gameObject.SetActive(false);
+        this.rewardParent.gameObject.SetActive(true);
+        rewardController.Initialize();
+    }
+
```

Comment:
This looks like a strange setup. What are we exactly trying to accomplish here? affinityParent is probably enabled for a single frame.

Reply:
Both the affinity and the rewards panels are in the same scenes. This function is switching between the two. It shouldn't be a problem, because Unity is single threaded so it will never "show" for a single frame.